lib/evmone/eof.cpp

```diff
@@ -104,10 +103,11 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_section_headers(byt
         {
             section_id = *it++;
 
-            // If DATA_SECTION is expected, CONTAINER_SECTION is also allowed, because
-            // container section is optional.
-            if (section_id != expected_section_id &&
-                (expected_section_id != CONTAINER_SECTION || section_id != DATA_SECTION))
+            // Skip optional sections.
+            if (section_id != expected_section_id && expected_section_id == CONTAINER_SECTION)
```

Comment:
matter of taste, but this reads weirder to me, weirder than the `get_section_missing_error` modification, and weirder than the "2nd" alternative which checks `CONTAINER_SECTION.size()` (which also keeps `get_section_missing_error` intact). The weirdness comes from the fact that the `expected_section_id` isn't a check-modify, but a check-maybe modify-check-modify sequence on each iteration.

Maybe we need a 3rd opinion to pick the least obtrusive fix, @gumb0 ?

Reply:
I think definitely this one here is more understandable.